### ISSUE TITLE
aio + docs/api: remove single character param names

### ISSUE THREAD

**Issue by petebacondarwin** [angular/angular#22137]
In https://github.com/angular/angular/pull/21874#issuecomment-363543097 @gkalpak suggested:

![image](https://user-images.githubusercontent.com/15655/36048523-cdb6b60c-0ddf-11e8-839d-17482305b5e1.png)

 I think it would be a good idea to have a check (via dgeni) that prevents [one-letter param names](https://pr21874-14230fd.ngbuilds.io/api/forms/Validators#nullvalidator)

**Comment by petebacondarwin** [comment]
I don't think that dgeni is the right place for such a check. It is really a code formatting issue.

**Comment by IgorMinar** [comment]
dgeni knows what apis are public, other tools don't have that visibility. where do you think this check should be if not in dgeni?

**Comment by petebacondarwin** [comment]
I thought the comment was a general one: that any function or method should not have single characters params?

If you are only interested in applying this to public APIs I can see that there is more reason to use Dgeni. I am just concerned that the time for such feedback is earlier for developers than when the docs are generated.

Your call. It is trivial to implement in dgeni.

**Comment by IgorMinar** [comment]
once this is done we can consider a tslint rule for all code, but as I said before that will take a long time to roll out so we shouldn't get blocked on that now.
